File: mobile/src/main/java/com/ourygo/ygomobile/util/SharedPreferenceUtil.java
package com.ourygo.ygomobile.util;

import static cn.garymb.ygomobile.Constants.PREF_IMMERSIVE_MODE;
import static cn.garymb.ygomobile.Constants.PREF_KEEP_SCALE;
import static cn.garymb.ygomobile.Constants.PREF_LOCK_SCREEN;
import static cn.garymb.ygomobile.Constants.PREF_OPENGL_VERSION;
import static cn.garymb.ygomobile.Constants.PREF_READ_EX;
import static cn.garymb.ygomobile.Constants.PREF_WINDOW_TOP_BOTTOM;

import android.content.SharedPreferences;

import cn.garymb.ygomobile.App;
import cn.garymb.ygomobile.AppsSettings;
import cn.garymb.ygomobile.lite.R;

public class SharedPreferenceUtil {

    public static final int DECK_EDIT_TYPE_LOCAL = 0;
    public static final int DECK_EDIT_TYPE_DECK_MANAGEMENT = 1;
    public static final int DECK_EDIT_TYPE_OURYGO_EZ = 2;

    public static final int SERVER_LIST_TYPE_LIST=0;
    public static final int SERVER_LIST_TYPE_GRID=1;


    //获取存放路径的share
    public static SharedPreferences getSharePath() {
        return App.get().getSharedPreferences("path", App.get().MODE_PRIVATE);
    }

    //获取存放类型的share
    public static SharedPreferences getShareType() {
        return App.get().getSharedPreferences("type", App.get().MODE_PRIVATE);
    }

    //获取存放开关状态的share
    public static SharedPreferences getShareKaiguan() {
        return App.get().getSharedPreferences("kaiguan", App.get().MODE_PRIVATE);
    }

    //获取各种记录的share
    public static SharedPreferences getShareRecord() {
        return App.get().getSharedPreferences("record", App.get().MODE_PRIVATE);
    }

    public static boolean addAppStartTimes() {
        return getShareRecord().edit().putInt("StartTimes", getAppStartTimes() + 1).commit();
    }

    //获取应用的启动次数
    public static int getAppStartTimes() {
        return getShareRecord().getInt("StartTimes", 0);
    }

    public static String getUserName() {
        return getShareRecord().getString("userName", null);
    }

    public static String getUserPassword() {
        return getShareRecord().getString("userPassword", null);
    }

    public static String getUserAccount() {
        return getShareRecord().getString("userAccount", null);
    }

    public static String getHttpSessionId() {
        return getShareRecord().getString("sessionId", null);
    }

    public static boolean setHttpSessionId(String sessionid) {
        return getShareRecord().edit().putString("sessionId", sessionid).commit();
        // TODO: Implement this method
    }

    public static boolean setMyCardUserName(String mycardUserName) {
        return getShareRecord().edit().putString(Record.ARG_MC_NAME, mycardUserName).commit();
    }

    public static String getMyCardUserName() {
        return getShareRecord().getString(Record.ARG_MC_NAME, null);
    }

    public static boolean setUserName(String name) {
        return getShareRecord().edit().putString("userName", name).commit();
    }

    public static boolean setUserAccount(String account) {
        return getShareRecord().edit().putString("userAccount", account).commit();
    }

    public static boolean setUserPassword(String password) {
        return getShareRecord().edit().putString("userPassword", password).commit();
    }

    public static boolean setScreenPadding(String paddding) {
        return AppsSettings.get().getSharedPreferences().edit().putString(PREF_WINDOW_TOP_BOTTOM, paddding).commit();
    }

    public static boolean setScreenPadding(int position) {
        return setScreenPadding(OYUtil.getArray(R.array.screen_top_bottom_value)[position]);
    }

    public static boolean setReadExpansions(boolean isReadExpansions) {
        return AppsSettings.get().getSharedPreferences().edit().putBoolean(PREF_READ_EX, isReadExpansions).commit();
    }

    public static boolean setOpenglVersion(int position) {
        return setOpenglVersion(OYUtil.getArray(R.array.opengl_version_value)[position]);
    }

    public static boolean setOpenglVersion(String opengl) {
        return AppsSettings.get().getSharedPreferences().edit().putString(PREF_OPENGL_VERSION, opengl).commit();
    }

    public static int getScreenPaddingPosition() {
        String value = AppsSettings.get().getScreenPadding() + "";
        String[] valueList = OYUtil.getArray(R.array.screen_top_bottom_value);
        for (int i = 0; i < valueList.length; i++) {
            String s = valueList[i];
            if (s.equals(value))
                return i;
        }
        return -1;
    }

    public static int getOpenglVersionPosition() {
        String value = AppsSettings.get().getOpenglVersion() + "";
        String[] valueList = OYUtil.getArray(R.array.opengl_version_value);
        for (int i = 0; i < valueList.length; i++) {
            String s = valueList[i];
            if (s.equals(value))
                return i;
        }
        return -1;
    }

    public static boolean setImmersiveMode(boolean isImmeriveMode) {
        return AppsSettings.get().getSharedPreferences().edit().putBoolean(PREF_IMMERSIVE_MODE, isImmeriveMode).commit();
    }

    public static boolean setKeepScale(boolean isKeepScale) {
        return AppsSettings.get().getSharedPreferences().edit().putBoolean(PREF_KEEP_SCALE, isKeepScale).commit();
    }

    public static boolean setHorizontal(boolean isHorizontal) {
        return AppsSettings.get().getSharedPreferences().edit().putBoolean(PREF_LOCK_SCREEN, isHorizontal).commit();
    }

    public static boolean isShowEz() {
        return getShareKaiguan().getBoolean("isShowEz", true);
    }

    public static boolean setIsShowEz(boolean isShow) {
        return getShareKaiguan().edit().putBoolean("isShowEz", isShow).commit();
    }

    public static boolean isShowVisitDeck() {
        return getShareKaiguan().getBoolean("isShowVisitDeck", true);
    }

    public static boolean setShowVisitDeck(boolean isShow) {
        return getShareKaiguan().edit().putBoolean("isShowVisitDeck", isShow).commit();
    }

    public static boolean isFristStart() {
        return getShareRecord().getBoolean("isFirstStart", true);
    }

    public static boolean setFirstStart(boolean isFirstStart) {
        return getShareRecord().edit().putBoolean("isFirstStart", isFirstStart).commit();
    }

    public static int getNextAifadianNum() {
        return getShareRecord().getInt("nextAifadianNum", (10 + (int) (Math.random() * 20)));
    }

    public static void setNextAifadianNum(int num) {
        getShareRecord().edit().putInt("nextAifadianNum", num).apply();
    }

    public static int getDeckEditType() {
        return getShareType().getInt("deckEditType", DECK_EDIT_TYPE_DECK_MANAGEMENT);
    }

    public static void setDeckEditType(int type) {
        getShareType().edit().putInt("deckEditType", type).apply();
    }

    public static int getServerListType() {
        return getShareType().getInt("serverListMode", SERVER_LIST_TYPE_LIST);
    }

    public static void setServerListType(int type) {
        getShareType().edit().putInt("serverListMode", type).apply();
    }

    public static long getVersionUpdateTime() {
        return getShareRecord().getLong("versionUpdateTime", 0);
    }

    public static void setVersionUpdateTime(long versionUpdateTime) {
        getShareRecord().edit().putLong("versionUpdateTime", versionUpdateTime).apply();
    }

    public static boolean isToastNewCardBag() {
        return getShareRecord().getBoolean("isToastNewCardBag", true);
    }

    public static void setToastNewCardBag(boolean toastNewCardBag) {
        getShareRecord().edit().putBoolean("isToastNewCardBag", toastNewCardBag).apply();
    }

    public static void setTodayStartTime(long versionUpdateTime) {
        getShareRecord().edit().putLong("todayStartTime", versionUpdateTime).apply();
    }

    public static long getTodayStartTime() {
        return getShareRecord().getLong("todayStartTime", 0);
    }

}

File: mobile/src/main/java/com/ourygo/ygomobile/util/SharedPreferenceUtil.kt
package com.ourygo.ygomobile.util

import android.content.Context
import android.content.SharedPreferences
import cn.garymb.ygomobile.App
import cn.garymb.ygomobile.AppsSettings
import cn.garymb.ygomobile.Constants
import cn.garymb.ygomobile.lite.R

object SharedPreferenceUtil {
    const val DECK_EDIT_TYPE_LOCAL = 0
    const val DECK_EDIT_TYPE_DECK_MANAGEMENT = 1
    const val DECK_EDIT_TYPE_OURYGO_EZ = 2
    const val SERVER_LIST_TYPE_LIST = 0
    const val SERVER_LIST_TYPE_GRID = 1
    val sharePath: SharedPreferences
        //获取存放路径的share
        get() = App.get().getSharedPreferences("path", Context.MODE_PRIVATE)
    val shareType: SharedPreferences
        //获取存放类型的share
        get() = App.get().getSharedPreferences("type", Context.MODE_PRIVATE)
    val shareKaiguan: SharedPreferences
        //获取存放开关状态的share
        get() = App.get().getSharedPreferences("kaiguan", Context.MODE_PRIVATE)
    val shareRecord: SharedPreferences
        //获取各种记录的share
        get() = App.get().getSharedPreferences("record", Context.MODE_PRIVATE)

    fun addAppStartTimes(): Boolean {
        return shareRecord.edit().putInt("StartTimes", appStartTimes + 1).commit()
    }

    val appStartTimes: Int
        //获取应用的启动次数
        get() = shareRecord.getInt("StartTimes", 0)
    val userName: String?
        get() = shareRecord.getString("userName", null)
    val userPassword: String?
        get() = shareRecord.getString("userPassword", null)
    val userAccount: String?
        get() = shareRecord.getString("userAccount", null)
    val httpSessionId: String?
        get() = shareRecord.getString("sessionId", null)

    fun setHttpSessionId(sessionid: String?): Boolean {
        return shareRecord.edit().putString("sessionId", sessionid).commit()
        // TODO: Implement this method
    }

    fun setMyCardUserName(mycardUserName: String?): Boolean {
        return shareRecord.edit().putString(Record.ARG_MC_NAME, mycardUserName).commit()
    }

    val myCardUserName: String?
        get() = shareRecord.getString(Record.ARG_MC_NAME, null)

    fun setUserName(name: String?): Boolean {
        return shareRecord.edit().putString("userName", name).commit()
    }

    fun setUserAccount(account: String?): Boolean {
        return shareRecord.edit().putString("userAccount", account).commit()
    }

    fun setUserPassword(password: String?): Boolean {
        return shareRecord.edit().putString("userPassword", password).commit()
    }

    fun setScreenPadding(paddding: String?): Boolean {
        return AppsSettings.get().sharedPreferences.edit()
            .putString(Constants.PREF_WINDOW_TOP_BOTTOM, paddding).commit()
    }

    @JvmStatic
    fun setScreenPadding(position: Int): Boolean {
        return setScreenPadding(OYUtil.getArray(R.array.screen_top_bottom_value)[position])
    }

    fun setReadExpansions(isReadExpansions: Boolean): Boolean {
        return AppsSettings.get().sharedPreferences.edit()
            .putBoolean(Constants.PREF_READ_EX, isReadExpansions).commit()
    }

    @JvmStatic
    fun setOpenglVersion(position: Int): Boolean {
        return setOpenglVersion(OYUtil.getArray(R.array.opengl_version_value)[position])
    }

    fun setOpenglVersion(opengl: String?): Boolean {
        return AppsSettings.get().sharedPreferences.edit()
            .putString(Constants.PREF_OPENGL_VERSION, opengl).commit()
    }

    @JvmStatic
    val screenPaddingPosition: Int
        get() {
            val value = AppsSettings.get().screenPadding.toString() + ""
            val valueList = OYUtil.getArray(R.array.screen_top_bottom_value)
            for (i in valueList!!.indices) {
                val s = valueList[i]
                if (s == value) return i
            }
            return -1
        }
    @JvmStatic
    val openglVersionPosition: Int
        get() {
            val value = AppsSettings.get().openglVersion.toString() + ""
            val valueList = OYUtil.getArray(R.array.opengl_version_value)
            for (i in valueList!!.indices) {
                val s = valueList[i]
                if (s == value) return i
            }
            return -1
        }

    @JvmStatic
    fun setImmersiveMode(isImmeriveMode: Boolean): Boolean {
        return AppsSettings.get().sharedPreferences.edit()
            .putBoolean(Constants.PREF_IMMERSIVE_MODE, isImmeriveMode).commit()
    }

    @JvmStatic
    fun setKeepScale(isKeepScale: Boolean): Boolean {
        return AppsSettings.get().sharedPreferences.edit()
            .putBoolean(Constants.PREF_KEEP_SCALE, isKeepScale).commit()
    }

    @JvmStatic
    fun setHorizontal(isHorizontal: Boolean): Boolean {
        return AppsSettings.get().sharedPreferences.edit()
            .putBoolean(Constants.PREF_LOCK_SCREEN, isHorizontal).commit()
    }

    val isShowEz: Boolean
        get() = shareKaiguan.getBoolean("isShowEz", true)

    fun setIsShowEz(isShow: Boolean): Boolean {
        return shareKaiguan.edit().putBoolean("isShowEz", isShow).commit()
    }

    val isShowVisitDeck: Boolean
        get() = shareKaiguan.getBoolean("isShowVisitDeck", true)

    fun setShowVisitDeck(isShow: Boolean): Boolean {
        return shareKaiguan.edit().putBoolean("isShowVisitDeck", isShow).commit()
    }

    val isFristStart: Boolean
        get() = shareRecord.getBoolean("isFirstStart", true)

    fun setFirstStart(isFirstStart: Boolean): Boolean {
        return shareRecord.edit().putBoolean("isFirstStart", isFirstStart).commit()
    }

    var nextAifadianNum: Int
        get() = shareRecord.getInt("nextAifadianNum", 10 + (Math.random() * 20).toInt())
        set(num) {
            shareRecord.edit().putInt("nextAifadianNum", num).apply()
        }
    @JvmStatic
    var deckEditType: Int
        get() = shareType.getInt("deckEditType", DECK_EDIT_TYPE_DECK_MANAGEMENT)
        set(type) {
            shareType.edit().putInt("deckEditType", type).apply()
        }
    @JvmStatic
    var serverListType: Int
        get() = shareType.getInt("serverListMode", SERVER_LIST_TYPE_LIST)
        set(type) {
            shareType.edit().putInt("serverListMode", type).apply()
        }
    @JvmStatic
    var versionUpdateTime: Long
        get() = shareRecord.getLong("versionUpdateTime", 0)
        set(versionUpdateTime) {
            shareRecord.edit().putLong("versionUpdateTime", versionUpdateTime).apply()
        }
    @JvmStatic
    var isToastNewCardBag: Boolean
        get() = shareRecord.getBoolean("isToastNewCardBag", true)
        set(toastNewCardBag) {
            shareRecord.edit().putBoolean("isToastNewCardBag", toastNewCardBag).apply()
        }
    var todayStartTime: Long
        get() = shareRecord.getLong("todayStartTime", 0)
        set(versionUpdateTime) {
            shareRecord.edit().putLong("todayStartTime", versionUpdateTime).apply()
        }
}